convert_gp_lookup.R

```diff
@@ -0,0 +1,193 @@
+#####################################################
+# GP Lookup
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+
+## packages ##
+library(dplyr)
+library(tidyr)
+library(haven)
+library(phsopendata)
+library(janitor)
+library(fs)
+
+
+## output ##
+output <- haven::read_sav(get_slf_gpprac_path())
+
+
+
+## get data ##
+
+
+# use `00_get_gp_cluster_data.R`- draws from open data
+# latest update date
+latest_update <- "Dec_2021"
+
+# Retrieve the latest resource from the dataset
+opendata <-
+  get_dataset("gp-practice-contact-details-and-list-sizes",
+    max_resources = 1
+  ) %>%
+  clean_names() %>%
+  # Filter and save
+  select(
+    gpprac = practice_code,
+    practice_name = gp_practice_name,
+    postcode,
+    cluster = gp_cluster
+  ) %>%
+  # Sort for SPSS matching
+  arrange(gpprac) %>%
+  # Write as an SPSS file
+  write_sav(
+    path(
+      # lookup_dir,
+      get_lookups_dir(),
+      paste0("practice_details_", latest_update),
+      ext = "zsav"
+    ),
+    compress = TRUE
+  )
+
+
+# gp lookup
+gp <-
+  haven::read_sav("/conf/linkage/output/lookups/Unicode/National Reference Files/gpprac.sav") %>%
+  # select only praccode and postcode
+  select(
+    praccode,
+    postcode
+  ) %>%
+  # sort by praccode
+  arrange(praccode) %>%
+  # rename praccode to allow join
+  rename(gpprac = "praccode")
+
+
+
+## match cluster information onto the practice reference list ##
+data <-
+  opendata %>%
+  # join cluster and gp data
+  left_join(gp, by = c("gpprac", "postcode")) %>%
+  # sort by postcode
+  arrange(postcode)
+
+
+## match on geography info ##
+
+# postcode lookup
+pc <- read_spd_file() %>%
+  select(
+    pc7,
+    pc8,
+    hb2018,
+    hscp2018,
+    ca2018
+  ) %>%
+  # rename pc8
+  rename(postcode = "pc8")
+
+
+data <-
+  data %>%
+  # join data and postcode data
+  left_join(pc, by = "postcode") %>%
+  # rename hb2018
+  rename(hbpraccode = "hb2018") %>%
+  select(
+    gpprac,
+    pc7,
+    postcode,
+    cluster,
+    hbpraccode,
+    hscp2018,
+    ca2018
+  )
+
+
+## ca to lca code ##
+# function will just be called once PR approved
```

Comment:
I agree, this is okay for now as you have already created the function in another branch. Once it is approved, remember to bring this branch back in line with main-R before merging :)